test/test_reductions.py

```diff
@@ -2843,6 +2843,13 @@ def test_against_np(tensor, bins=100, min=0, max=0):
         expanded = torch.randn(1, 5, 1, 2, device=device).expand(3, 5, 7, 2)
         test_against_np(expanded)
 
+        if torch.cuda.is_available():
```

Comment:
would test_against_np catch it instead?

Reply:
Unfortunately, there is a discrepancy with numpy for this input tensor. 
The code 
```python
import torch
import numpy as np

def fn(x):
    return torch.histc(x, bins=10, min=0, max=0.99)

def fn_np(x):
    return torch.from_numpy(np.histogram(x.numpy(), bins=10, range=(0, 0.99))[0])


x = torch.linspace(0, 0.99, 1001, dtype=torch.float32)
torch.histc(x, bins=10, min=0, max=0.99)

print(fn(x))
print(fn(x.cuda()))
print(fn_np(x))
```
Produces:
```
tensor([101.,  99., 101.,  99., 101.,  99., 100., 100., 100., 101.])
tensor([101.,  99., 101.,  99., 101.,  99., 100., 100., 100., 101.], device='cuda:0')
tensor([101,  99, 100, 100, 100, 100, 100, 100, 100, 101])
```

Reply:
Is NumPy's `histogram` more accurate? Making `histc` consistent on CPU and CUDA is nice, but is the `histc` algorithm we're using actually correct?

Reply:
NumPy might be more accurate because it uses direct comparison https://github.com/numpy/numpy/blob/13d55a3c2f016a58a6e9d6b8086f338e07c7478f/numpy/lib/histograms.py#L862-L866 if the value is within the range or not. However, I noticed that it struggles with this case too as it produces different results for `float32` and `float64`.

Reply:
Instead of conditioning on CUDA being available, just compare the result to the expected outcome of 

```
tensor([101.,  99., 101.,  99., 101.,  99., 100., 100., 100., 101.])
```

directly, and add a comment that NumPy produces a different result

Reply:
If you change boundary computation to what cpu is doing now (instead of to what cuda is doing) would the results match?

Reply:
Yes, the results will exactly match if I slightly change the input tensor or histogram boundaries by a machine epsilon.